<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Track data input | BioSeqUtils Reference</title>
  <meta name="description" content="This is a detailed documentation of BioSeqUtils package." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Track data input | BioSeqUtils Reference" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a detailed documentation of BioSeqUtils package." />
  <meta name="github-repo" content="rstudio/bookdown-BioSeqUtils" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Track data input | BioSeqUtils Reference" />
  
  <meta name="twitter:description" content="This is a detailed documentation of BioSeqUtils package." />
  

<meta name="author" content="Jun Zhang" />


<meta name="date" content="2024-11-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="draw-chromosome-ideograms.html"/>
<link rel="next" href="basic-visualization.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">BioSeqUtils manual documentation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#related-blogs"><i class="fa fa-check"></i><b>1.1</b> Related blogs</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="extract-sequence.html"><a href="extract-sequence.html"><i class="fa fa-check"></i><b>2</b> Extract sequence</a>
<ul>
<li class="chapter" data-level="2.1" data-path="extract-sequence.html"><a href="extract-sequence.html#data-loading"><i class="fa fa-check"></i><b>2.1</b> Data loading</a></li>
<li class="chapter" data-level="2.2" data-path="extract-sequence.html"><a href="extract-sequence.html#get-features-information"><i class="fa fa-check"></i><b>2.2</b> Get features information</a></li>
<li class="chapter" data-level="2.3" data-path="extract-sequence.html"><a href="extract-sequence.html#extract-sequnence"><i class="fa fa-check"></i><b>2.3</b> Extract sequnence</a></li>
<li class="chapter" data-level="2.4" data-path="extract-sequence.html"><a href="extract-sequence.html#get-inrton-information"><i class="fa fa-check"></i><b>2.4</b> Get inrton information</a></li>
<li class="chapter" data-level="2.5" data-path="extract-sequence.html"><a href="extract-sequence.html#extract-promoters"><i class="fa fa-check"></i><b>2.5</b> Extract promoters</a></li>
<li class="chapter" data-level="2.6" data-path="extract-sequence.html"><a href="extract-sequence.html#extract-non-redundant-transcript-length"><i class="fa fa-check"></i><b>2.6</b> Extract non-redundant transcript length</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="draw-chromosome-ideograms.html"><a href="draw-chromosome-ideograms.html"><i class="fa fa-check"></i><b>3</b> Draw chromosome ideograms</a>
<ul>
<li class="chapter" data-level="3.1" data-path="draw-chromosome-ideograms.html"><a href="draw-chromosome-ideograms.html#downloading-related-data"><i class="fa fa-check"></i><b>3.1</b> Downloading related data</a></li>
<li class="chapter" data-level="3.2" data-path="draw-chromosome-ideograms.html"><a href="draw-chromosome-ideograms.html#making-plot"><i class="fa fa-check"></i><b>3.2</b> Making plot</a></li>
<li class="chapter" data-level="3.3" data-path="draw-chromosome-ideograms.html"><a href="draw-chromosome-ideograms.html#drawing-gene-density-manually"><i class="fa fa-check"></i><b>3.3</b> Drawing gene density manually</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="track-data-input.html"><a href="track-data-input.html"><i class="fa fa-check"></i><b>4</b> Track data input</a>
<ul>
<li class="chapter" data-level="4.1" data-path="track-data-input.html"><a href="track-data-input.html#load-signal-data"><i class="fa fa-check"></i><b>4.1</b> Load signal data</a></li>
<li class="chapter" data-level="4.2" data-path="track-data-input.html"><a href="track-data-input.html#load-peaks-data"><i class="fa fa-check"></i><b>4.2</b> Load peaks data</a></li>
<li class="chapter" data-level="4.3" data-path="track-data-input.html"><a href="track-data-input.html#load-links-data"><i class="fa fa-check"></i><b>4.3</b> Load links data</a></li>
<li class="chapter" data-level="4.4" data-path="track-data-input.html"><a href="track-data-input.html#load-hic-related-data"><i class="fa fa-check"></i><b>4.4</b> Load Hic related data</a></li>
<li class="chapter" data-level="4.5" data-path="track-data-input.html"><a href="track-data-input.html#extract-junction-data"><i class="fa fa-check"></i><b>4.5</b> Extract junction data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="basic-visualization.html"><a href="basic-visualization.html"><i class="fa fa-check"></i><b>5</b> Basic visualization</a>
<ul>
<li class="chapter" data-level="5.1" data-path="basic-visualization.html"><a href="basic-visualization.html#basic-plot"><i class="fa fa-check"></i><b>5.1</b> Basic plot</a></li>
<li class="chapter" data-level="5.2" data-path="basic-visualization.html"><a href="basic-visualization.html#signal-range-settings"><i class="fa fa-check"></i><b>5.2</b> Signal range settings</a></li>
<li class="chapter" data-level="5.3" data-path="basic-visualization.html"><a href="basic-visualization.html#transcript-track-settings"><i class="fa fa-check"></i><b>5.3</b> Transcript track settings</a></li>
<li class="chapter" data-level="5.4" data-path="basic-visualization.html"><a href="basic-visualization.html#intron-line-type"><i class="fa fa-check"></i><b>5.4</b> Intron line type</a></li>
<li class="chapter" data-level="5.5" data-path="basic-visualization.html"><a href="basic-visualization.html#x-axis-limitation"><i class="fa fa-check"></i><b>5.5</b> X axis limitation</a></li>
<li class="chapter" data-level="5.6" data-path="basic-visualization.html"><a href="basic-visualization.html#background-and-highlight-region-settings"><i class="fa fa-check"></i><b>5.6</b> Background and highlight region settings</a></li>
<li class="chapter" data-level="5.7" data-path="basic-visualization.html"><a href="basic-visualization.html#order-for-genes-and-samples"><i class="fa fa-check"></i><b>5.7</b> Order for genes and samples</a></li>
<li class="chapter" data-level="5.8" data-path="basic-visualization.html"><a href="basic-visualization.html#adding-group-information-for-genes-and-samples"><i class="fa fa-check"></i><b>5.8</b> Adding group information for genes and samples</a></li>
<li class="chapter" data-level="5.9" data-path="basic-visualization.html"><a href="basic-visualization.html#strip-settings"><i class="fa fa-check"></i><b>5.9</b> Strip settings</a></li>
<li class="chapter" data-level="5.10" data-path="basic-visualization.html"><a href="basic-visualization.html#draw-chromosome-annotation"><i class="fa fa-check"></i><b>5.10</b> Draw chromosome annotation</a></li>
<li class="chapter" data-level="5.11" data-path="basic-visualization.html"><a href="basic-visualization.html#plot-with-specified-genomic-region"><i class="fa fa-check"></i><b>5.11</b> Plot with specified genomic region</a></li>
<li class="chapter" data-level="5.12" data-path="basic-visualization.html"><a href="basic-visualization.html#adding-peaks-track"><i class="fa fa-check"></i><b>5.12</b> Adding peaks track</a></li>
<li class="chapter" data-level="5.13" data-path="basic-visualization.html"><a href="basic-visualization.html#extending-bases-upstream-and-downstream"><i class="fa fa-check"></i><b>5.13</b> Extending bases upstream and downstream</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="loops-visualization.html"><a href="loops-visualization.html"><i class="fa fa-check"></i><b>6</b> Loops visualization</a>
<ul>
<li class="chapter" data-level="6.1" data-path="loops-visualization.html"><a href="loops-visualization.html#input-data-prepration"><i class="fa fa-check"></i><b>6.1</b> Input data prepration</a></li>
<li class="chapter" data-level="6.2" data-path="loops-visualization.html"><a href="loops-visualization.html#basic-loops-visualization"><i class="fa fa-check"></i><b>6.2</b> Basic loops visualization</a></li>
<li class="chapter" data-level="6.3" data-path="loops-visualization.html"><a href="loops-visualization.html#reverse-y-axis"><i class="fa fa-check"></i><b>6.3</b> Reverse Y axis</a></li>
<li class="chapter" data-level="6.4" data-path="loops-visualization.html"><a href="loops-visualization.html#combing-with-chromosome-track"><i class="fa fa-check"></i><b>6.4</b> Combing with chromosome track</a></li>
<li class="chapter" data-level="6.5" data-path="loops-visualization.html"><a href="loops-visualization.html#higlight-regions"><i class="fa fa-check"></i><b>6.5</b> Higlight regions</a></li>
<li class="chapter" data-level="6.6" data-path="loops-visualization.html"><a href="loops-visualization.html#drawing-only-loop-tracks"><i class="fa fa-check"></i><b>6.6</b> Drawing only loop tracks</a></li>
<li class="chapter" data-level="6.7" data-path="loops-visualization.html"><a href="loops-visualization.html#curve-geoms-choosing"><i class="fa fa-check"></i><b>6.7</b> Curve geoms choosing</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="hic-heatmap-visualization.html"><a href="hic-heatmap-visualization.html"><i class="fa fa-check"></i><b>7</b> Hic heatmap visualization</a>
<ul>
<li class="chapter" data-level="7.1" data-path="hic-heatmap-visualization.html"><a href="hic-heatmap-visualization.html#data-preperation"><i class="fa fa-check"></i><b>7.1</b> Data preperation</a></li>
<li class="chapter" data-level="7.2" data-path="hic-heatmap-visualization.html"><a href="hic-heatmap-visualization.html#basic-plot-1"><i class="fa fa-check"></i><b>7.2</b> Basic plot</a></li>
<li class="chapter" data-level="7.3" data-path="hic-heatmap-visualization.html"><a href="hic-heatmap-visualization.html#real-data-practice"><i class="fa fa-check"></i><b>7.3</b> Real data practice</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="splice-junction-visualization.html"><a href="splice-junction-visualization.html"><i class="fa fa-check"></i><b>8</b> Splice junction visualization</a>
<ul>
<li class="chapter" data-level="8.1" data-path="splice-junction-visualization.html"><a href="splice-junction-visualization.html#data-pareparation"><i class="fa fa-check"></i><b>8.1</b> Data pareparation</a></li>
<li class="chapter" data-level="8.2" data-path="splice-junction-visualization.html"><a href="splice-junction-visualization.html#visualization"><i class="fa fa-check"></i><b>8.2</b> Visualization</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="other-funs.html"><a href="other-funs.html"><i class="fa fa-check"></i><b>9</b> Other funs</a>
<ul>
<li class="chapter" data-level="9.1" data-path="other-funs.html"><a href="other-funs.html#transcript-coordinate-transformation"><i class="fa fa-check"></i><b>9.1</b> Transcript coordinate transformation</a></li>
<li class="chapter" data-level="9.2" data-path="other-funs.html"><a href="other-funs.html#track-overlap"><i class="fa fa-check"></i><b>9.2</b> Track overlap</a></li>
<li class="chapter" data-level="9.3" data-path="other-funs.html"><a href="other-funs.html#extend-coordinate-for-gene"><i class="fa fa-check"></i><b>9.3</b> Extend coordinate for gene</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="frequntlly-asked-questions.html"><a href="frequntlly-asked-questions.html"><i class="fa fa-check"></i><b>10</b> Frequntlly asked questions</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BioSeqUtils Reference</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="track-data-input" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Track data input<a href="track-data-input.html#track-data-input" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>There are many tools or softwares to visualize NGS(Next Generation Sequencing) data including
<strong>ChIP-SEQ</strong>, <strong>ATAC-SEQ</strong>, <strong>RNA-SEQ</strong>, <strong>Hic</strong>, <strong>HiChIP</strong> and so on. The R packages <a href="https://bioconductor.org/packages/release/bioc/html/Gviz.html"><strong>Gviz</strong></a>, <a href="https://github.com/PhanstielLab/plotgardener"><strong>plotgardener</strong></a>, <a href="https://github.com/showteeth/ggcoverage"><strong>ggcoverage</strong></a> and <a href="https://www.bioconductor.org/packages/release/bioc/html/ggbio.html"><strong>ggbio</strong></a> are some popular tools
to visualize NGS data in R. Besides, the online or local softwares like <a href="https://software.broadinstitute.org/software/igv/download"><strong>IGV</strong></a>, <a href="http://epigenomegateway.wustl.edu/legacy/"><strong>Wubrowse</strong></a> and <a href="https://genome.ucsc.edu/"><strong>ucsc genome browser</strong></a>. There are still some limitations for some of them to
make a <strong>High-Quality Graphics for Publication</strong> with less code and less time.
Here I supply some utilities and functions totally based on <strong>ggplot2</strong> package to visualize
NGS data in R and create a nice graph with high-quality for publication.</p>
<p>Though I have written a similar package <a href="https://github.com/junjunlab/transPlotR"><strong>transPlotR</strong></a> which plays some same roles on generating tracks plot. Some limitations and shortcomings come
to me when I use this package. So I devote myself again to develop and expand the functions.</p>
<p>The main plot function is <strong>trackVisProMax</strong> function and this is combined with some
data load input functions for create a track graph.</p>
<div id="load-signal-data" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Load signal data<a href="track-data-input.html#load-signal-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>bigwig</strong> format is binary of <strong>wig</strong> which’s data size is samller. I recommend use this
format, <strong>“wig”</strong> and <strong>“bedGraph”</strong> format are also accepted, you should just define <strong>format</strong>
parameter. <strong>loadBigWig</strong> function can read bigwig data into R which based on <a href="https://bioconductor.org/packages/release/bioc/html/rtracklayer.html"><strong>rtracklayer::import.bw</strong></a>.
Here are some examples to load your bigwig data:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="track-data-input.html#cb15-1" tabindex="-1"></a><span class="fu">library</span>(BioSeqUtils)</span>
<span id="cb15-2"><a href="track-data-input.html#cb15-2" tabindex="-1"></a></span>
<span id="cb15-3"><a href="track-data-input.html#cb15-3" tabindex="-1"></a><span class="co"># load bigwig files</span></span>
<span id="cb15-4"><a href="track-data-input.html#cb15-4" tabindex="-1"></a>file <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">&quot;test-bw/&quot;</span>,<span class="at">pattern =</span> <span class="st">&#39;.bw&#39;</span>,<span class="at">full.names =</span> T)</span>
<span id="cb15-5"><a href="track-data-input.html#cb15-5" tabindex="-1"></a>file</span>
<span id="cb15-6"><a href="track-data-input.html#cb15-6" tabindex="-1"></a><span class="co"># [1] &quot;test-bw/1cell-m6A-1.bw&quot; &quot;test-bw/1cell-m6A-2.bw&quot; &quot;test-bw/1cell-RNA-1.bw&quot;</span></span>
<span id="cb15-7"><a href="track-data-input.html#cb15-7" tabindex="-1"></a><span class="co"># [4] &quot;test-bw/1cell-RNA-2.bw&quot; &quot;test-bw/2cell-m6A-1.bw&quot; &quot;test-bw/2cell-m6A-2.bw&quot;</span></span>
<span id="cb15-8"><a href="track-data-input.html#cb15-8" tabindex="-1"></a><span class="co"># [7] &quot;test-bw/2cell-RNA-1.bw&quot; &quot;test-bw/2cell-RNA-2.bw&quot;</span></span>
<span id="cb15-9"><a href="track-data-input.html#cb15-9" tabindex="-1"></a></span>
<span id="cb15-10"><a href="track-data-input.html#cb15-10" tabindex="-1"></a><span class="co"># select some chromosomes for test</span></span>
<span id="cb15-11"><a href="track-data-input.html#cb15-11" tabindex="-1"></a>bw <span class="ot">&lt;-</span> <span class="fu">loadBigWig</span>(<span class="at">bw_file =</span> file,<span class="at">chrom =</span> <span class="fu">c</span>(<span class="st">&quot;5&quot;</span>,<span class="st">&quot;15&quot;</span>),<span class="at">format =</span> <span class="st">&quot;bw&quot;</span>)</span>
<span id="cb15-12"><a href="track-data-input.html#cb15-12" tabindex="-1"></a></span>
<span id="cb15-13"><a href="track-data-input.html#cb15-13" tabindex="-1"></a><span class="co"># check</span></span>
<span id="cb15-14"><a href="track-data-input.html#cb15-14" tabindex="-1"></a><span class="fu">head</span>(bw,<span class="dv">3</span>)</span>
<span id="cb15-15"><a href="track-data-input.html#cb15-15" tabindex="-1"></a><span class="co">#   seqnames   start     end   score    fileName</span></span>
<span id="cb15-16"><a href="track-data-input.html#cb15-16" tabindex="-1"></a><span class="co"># 1       15       1 3054635 0.00000 1cell-m6A-1</span></span>
<span id="cb15-17"><a href="track-data-input.html#cb15-17" tabindex="-1"></a><span class="co"># 2       15 3054636 3054640 1.34079 1cell-m6A-1</span></span>
<span id="cb15-18"><a href="track-data-input.html#cb15-18" tabindex="-1"></a><span class="co"># 3       15 3054641 3054715 2.68159 1cell-m6A-1</span></span></code></pre></div>
<p>For saving space, we selected <strong><em>chromosome 5</em></strong> and <strong><em>chromosome 15</em></strong> for each file, if you don’t
specify <strong>chrom</strong> parameter, <strong>loadBigWig</strong> will return all chromosomes. You can
also specify <strong>file_name</strong> to assign a new name for your each bigwig data.</p>
</div>
<div id="load-peaks-data" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Load peaks data<a href="track-data-input.html#load-peaks-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>loadBed</strong> function allows you to read peaks data in R which ias baesd on <a href="https://bioconductor.org/packages/release/bioc/html/rtracklayer.html"><strong>rtracklayer::import.bed</strong></a>.
Usually we will select the first three columns for downstream analysis. You can
also specify <strong>file_name</strong> to assign a new name for your each bed data. Here is
a example to read peaks data:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="track-data-input.html#cb16-1" tabindex="-1"></a>bedfile <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">&quot;./&quot;</span>,<span class="at">pattern =</span> <span class="st">&quot;.bed&quot;</span>)</span>
<span id="cb16-2"><a href="track-data-input.html#cb16-2" tabindex="-1"></a><span class="co"># [1] &quot;peaks.bed&quot;  &quot;peaks2.bed&quot;</span></span>
<span id="cb16-3"><a href="track-data-input.html#cb16-3" tabindex="-1"></a></span>
<span id="cb16-4"><a href="track-data-input.html#cb16-4" tabindex="-1"></a>bed_df <span class="ot">&lt;-</span> <span class="fu">loadBed</span>(bedfile)</span>
<span id="cb16-5"><a href="track-data-input.html#cb16-5" tabindex="-1"></a></span>
<span id="cb16-6"><a href="track-data-input.html#cb16-6" tabindex="-1"></a><span class="co"># check</span></span>
<span id="cb16-7"><a href="track-data-input.html#cb16-7" tabindex="-1"></a><span class="fu">head</span>(bed_df,<span class="dv">3</span>)</span>
<span id="cb16-8"><a href="track-data-input.html#cb16-8" tabindex="-1"></a><span class="co">#   seqnames     start       end sampleName y</span></span>
<span id="cb16-9"><a href="track-data-input.html#cb16-9" tabindex="-1"></a><span class="co"># 1        5 142905501 142905600      peaks 1</span></span>
<span id="cb16-10"><a href="track-data-input.html#cb16-10" tabindex="-1"></a><span class="co"># 2        5 142903201 142903800      peaks 1</span></span>
<span id="cb16-11"><a href="track-data-input.html#cb16-11" tabindex="-1"></a><span class="co"># 3       15  61985342  61985900      peaks 1</span></span></code></pre></div>
</div>
<div id="load-links-data" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Load links data<a href="track-data-input.html#load-links-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Links data often describe two interaction sites on genomic positions. <strong>Hic</strong>
and <strong>HiChIP</strong> technologies can achive this goal. The data format can be <strong>bed</strong>
and <strong>bedpe</strong> format. Or you can supply with only 4 columns(<strong>chrom</strong>, <strong>start</strong>,
<strong>end</strong>, <strong>value</strong>) format data. <strong>loadloops</strong> function allows you to read these format
data in R. Example shows in the following code:</p>
<blockquote>
<p><strong><em>Note:</em></strong> You should supply <strong>file_name</strong> for each file.</p>
</blockquote>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="track-data-input.html#cb17-1" tabindex="-1"></a>loop_file <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;test-bw2/&quot;</span>,<span class="at">pattern =</span> <span class="st">&quot;.bedpe$&quot;</span>,<span class="at">full.names =</span> T)</span>
<span id="cb17-2"><a href="track-data-input.html#cb17-2" tabindex="-1"></a>loop_file</span>
<span id="cb17-3"><a href="track-data-input.html#cb17-3" tabindex="-1"></a><span class="co"># [1] &quot;test-bw2/C1-CTCF.bedpe&quot;    &quot;test-bw2/C1-H3K27ac.bedpe&quot; &quot;test-bw2/M1-CTCF.bedpe&quot;   </span></span>
<span id="cb17-4"><a href="track-data-input.html#cb17-4" tabindex="-1"></a><span class="co"># [4] &quot;test-bw2/M1-H3K27ac.bedpe&quot;</span></span>
<span id="cb17-5"><a href="track-data-input.html#cb17-5" tabindex="-1"></a></span>
<span id="cb17-6"><a href="track-data-input.html#cb17-6" tabindex="-1"></a>file_name <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;C1-CTCF&quot;</span>,<span class="st">&quot;C1-H3K27ac&quot;</span>,<span class="st">&quot;M1-CTCF&quot;</span>,<span class="st">&quot;M1-H3K27ac&quot;</span>)</span>
<span id="cb17-7"><a href="track-data-input.html#cb17-7" tabindex="-1"></a></span>
<span id="cb17-8"><a href="track-data-input.html#cb17-8" tabindex="-1"></a><span class="co"># test code</span></span>
<span id="cb17-9"><a href="track-data-input.html#cb17-9" tabindex="-1"></a>loop_data <span class="ot">&lt;-</span> <span class="fu">loadloops</span>(<span class="at">loop_file =</span> loop_file,<span class="at">file_name =</span> file_name,</span>
<span id="cb17-10"><a href="track-data-input.html#cb17-10" tabindex="-1"></a>                       <span class="at">sep =</span> <span class="st">&quot; &quot;</span>)</span>
<span id="cb17-11"><a href="track-data-input.html#cb17-11" tabindex="-1"></a></span>
<span id="cb17-12"><a href="track-data-input.html#cb17-12" tabindex="-1"></a><span class="co"># check</span></span>
<span id="cb17-13"><a href="track-data-input.html#cb17-13" tabindex="-1"></a><span class="fu">head</span>(loop_data,<span class="dv">3</span>)</span>
<span id="cb17-14"><a href="track-data-input.html#cb17-14" tabindex="-1"></a></span>
<span id="cb17-15"><a href="track-data-input.html#cb17-15" tabindex="-1"></a><span class="co">#   seqnames     start       end    score fileName</span></span>
<span id="cb17-16"><a href="track-data-input.html#cb17-16" tabindex="-1"></a><span class="co"># 1    chr10 100002774 100022436 0.021354  C1-CTCF</span></span>
<span id="cb17-17"><a href="track-data-input.html#cb17-17" tabindex="-1"></a><span class="co"># 2    chr10 100002774 100069170 0.068404  C1-CTCF</span></span>
<span id="cb17-18"><a href="track-data-input.html#cb17-18" tabindex="-1"></a><span class="co"># 3    chr10 100002774 100185646 0.184670  C1-CTCF</span></span></code></pre></div>
</div>
<div id="load-hic-related-data" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Load Hic related data<a href="track-data-input.html#load-hic-related-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Hic related data format is multiple includes <strong>.h5</strong>, <strong>.hic</strong>,
<strong>.cool</strong>, <strong>.mcool</strong> and so on. It mainly depends on what kind of
soft/tool in the upstream. The usual format is <strong>.hic</strong> and <strong>.cool</strong>.
So we foucus on these fromat for visualization. You can use <strong>hicConvertFormat</strong>
command in <a href="https://hicexplorer.readthedocs.io/en/latest/index.html"><strong>HiCExplorer</strong></a>
software to covert into suitable format if you have other format file.</p>
<p>Different upstream tools will generate different resolution matrix data,
please make sure what resolution the Hic data you are using. For <strong>.hic</strong>
data, you can use <a href="https://github.com/cran/strawr"><strong>strawr::readHicBpResolutions</strong></a>
function to check avaliable resolutions to use, examples shows here:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="track-data-input.html#cb18-1" tabindex="-1"></a><span class="co"># install.packages(&quot;strawr&quot;)</span></span>
<span id="cb18-2"><a href="track-data-input.html#cb18-2" tabindex="-1"></a><span class="fu">library</span>(strawr)</span>
<span id="cb18-3"><a href="track-data-input.html#cb18-3" tabindex="-1"></a></span>
<span id="cb18-4"><a href="track-data-input.html#cb18-4" tabindex="-1"></a><span class="co"># test data</span></span>
<span id="cb18-5"><a href="track-data-input.html#cb18-5" tabindex="-1"></a><span class="fu">readHicBpResolutions</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test.hic&quot;</span>, <span class="at">package =</span> <span class="st">&quot;strawr&quot;</span>))</span>
<span id="cb18-6"><a href="track-data-input.html#cb18-6" tabindex="-1"></a><span class="co"># [1] 2500000</span></span>
<span id="cb18-7"><a href="track-data-input.html#cb18-7" tabindex="-1"></a></span>
<span id="cb18-8"><a href="track-data-input.html#cb18-8" tabindex="-1"></a><span class="co"># real data</span></span>
<span id="cb18-9"><a href="track-data-input.html#cb18-9" tabindex="-1"></a><span class="fu">readHicBpResolutions</span>(<span class="st">&quot;test-bw2/RPE-ICRF193_5uM.hic&quot;</span>)</span>
<span id="cb18-10"><a href="track-data-input.html#cb18-10" tabindex="-1"></a><span class="co"># [1] 2500000 1000000  500000  250000  100000   50000   25000   10000    5000</span></span></code></pre></div>
<p><strong>readHicChroms</strong> function can be used to vies chromosome names for your data:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="track-data-input.html#cb19-1" tabindex="-1"></a><span class="co"># test data</span></span>
<span id="cb19-2"><a href="track-data-input.html#cb19-2" tabindex="-1"></a><span class="fu">readHicChroms</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test.hic&quot;</span>, <span class="at">package =</span> <span class="st">&quot;strawr&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-3"><a href="track-data-input.html#cb19-3" tabindex="-1"></a>  <span class="fu">head</span>()</span>
<span id="cb19-4"><a href="track-data-input.html#cb19-4" tabindex="-1"></a></span>
<span id="cb19-5"><a href="track-data-input.html#cb19-5" tabindex="-1"></a><span class="co">#   name    length</span></span>
<span id="cb19-6"><a href="track-data-input.html#cb19-6" tabindex="-1"></a><span class="co"># 1    1 249250621</span></span>
<span id="cb19-7"><a href="track-data-input.html#cb19-7" tabindex="-1"></a><span class="co"># 2   10 135534747</span></span>
<span id="cb19-8"><a href="track-data-input.html#cb19-8" tabindex="-1"></a><span class="co"># 3   11 135006516</span></span>
<span id="cb19-9"><a href="track-data-input.html#cb19-9" tabindex="-1"></a><span class="co"># 4   12 133851895</span></span>
<span id="cb19-10"><a href="track-data-input.html#cb19-10" tabindex="-1"></a><span class="co"># 5   13 115169878</span></span>
<span id="cb19-11"><a href="track-data-input.html#cb19-11" tabindex="-1"></a><span class="co"># 6   14 107349540</span></span>
<span id="cb19-12"><a href="track-data-input.html#cb19-12" tabindex="-1"></a></span>
<span id="cb19-13"><a href="track-data-input.html#cb19-13" tabindex="-1"></a><span class="co"># real data</span></span>
<span id="cb19-14"><a href="track-data-input.html#cb19-14" tabindex="-1"></a><span class="fu">readHicChroms</span>(<span class="st">&quot;test-bw2/RPE-ICRF193_5uM.hic&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span>
<span id="cb19-15"><a href="track-data-input.html#cb19-15" tabindex="-1"></a><span class="co">#   name    length</span></span>
<span id="cb19-16"><a href="track-data-input.html#cb19-16" tabindex="-1"></a><span class="co"># 1    1 249250621</span></span>
<span id="cb19-17"><a href="track-data-input.html#cb19-17" tabindex="-1"></a><span class="co"># 2   10 135534747</span></span>
<span id="cb19-18"><a href="track-data-input.html#cb19-18" tabindex="-1"></a><span class="co"># 3   11 135006516</span></span>
<span id="cb19-19"><a href="track-data-input.html#cb19-19" tabindex="-1"></a><span class="co"># 4   12 133851895</span></span>
<span id="cb19-20"><a href="track-data-input.html#cb19-20" tabindex="-1"></a><span class="co"># 5   13 115169878</span></span>
<span id="cb19-21"><a href="track-data-input.html#cb19-21" tabindex="-1"></a><span class="co"># 6   14 107349540</span></span></code></pre></div>
<p><strong>prepareHic</strong> function allows you to read <strong>.hic</strong> and <strong>.cool</strong> data.
Usually these data are big size and will spend a lot of space and memory.
Please making sure you have enough soures to deal with it. Here are some
eamples:</p>
<hr />
<blockquote>
<p>Test data can be fetched on <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE200160"><strong>GSE200160</strong></a>.</p>
</blockquote>
<p>For <strong><em>.hic</em></strong> data, we use <strong>plotgardener::readHic</strong> function to read data into R
and it is has been <strong>upper-triangle</strong> matrix format.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="track-data-input.html#cb20-1" tabindex="-1"></a>hic_data <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;test-bw2/&quot;</span>,<span class="at">pattern =</span> <span class="st">&quot;.hic&quot;</span>,<span class="at">full.names =</span> T)</span>
<span id="cb20-2"><a href="track-data-input.html#cb20-2" tabindex="-1"></a>hic_data</span>
<span id="cb20-3"><a href="track-data-input.html#cb20-3" tabindex="-1"></a><span class="co"># [1] &quot;test-bw2/RPE-doxorubicin_02uM.hic&quot;  &quot;test-bw2/RPE-doxorubicin_034uM.hic&quot;</span></span>
<span id="cb20-4"><a href="track-data-input.html#cb20-4" tabindex="-1"></a><span class="co"># [3] &quot;test-bw2/RPE-ICRF193_5uM.hic&quot;</span></span>
<span id="cb20-5"><a href="track-data-input.html#cb20-5" tabindex="-1"></a></span>
<span id="cb20-6"><a href="track-data-input.html#cb20-6" tabindex="-1"></a>hic_df <span class="ot">&lt;-</span> <span class="fu">prepareHic</span>(<span class="at">hic_path =</span> hic_data,</span>
<span id="cb20-7"><a href="track-data-input.html#cb20-7" tabindex="-1"></a>                     <span class="at">file_name =</span> <span class="fu">c</span>(<span class="st">&quot;doxorubicin_02uM&quot;</span>,<span class="st">&quot;doxorubicin_34uM&quot;</span>,</span>
<span id="cb20-8"><a href="track-data-input.html#cb20-8" tabindex="-1"></a>                                   <span class="st">&quot;ICRF193_5uM&quot;</span>),</span>
<span id="cb20-9"><a href="track-data-input.html#cb20-9" tabindex="-1"></a>                     <span class="at">chrom =</span> <span class="st">&quot;1&quot;</span>,<span class="at">assembly =</span> <span class="st">&quot;hg19&quot;</span>,</span>
<span id="cb20-10"><a href="track-data-input.html#cb20-10" tabindex="-1"></a>                     <span class="at">resolution =</span> <span class="dv">10000</span>)</span>
<span id="cb20-11"><a href="track-data-input.html#cb20-11" tabindex="-1"></a></span>
<span id="cb20-12"><a href="track-data-input.html#cb20-12" tabindex="-1"></a><span class="co"># check</span></span>
<span id="cb20-13"><a href="track-data-input.html#cb20-13" tabindex="-1"></a><span class="fu">head</span>(hic_df[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,])</span>
<span id="cb20-14"><a href="track-data-input.html#cb20-14" tabindex="-1"></a><span class="co">#   seqnames  start   end     score         fileName id</span></span>
<span id="cb20-15"><a href="track-data-input.html#cb20-15" tabindex="-1"></a><span class="co"># 1     chr1 710000  5000  72.90664 doxorubicin_02uM  1</span></span>
<span id="cb20-16"><a href="track-data-input.html#cb20-16" tabindex="-1"></a><span class="co"># 2     chr1 715000     0 323.53738 doxorubicin_02uM  2</span></span>
<span id="cb20-17"><a href="track-data-input.html#cb20-17" tabindex="-1"></a><span class="co"># 3     chr1 715000 10000 212.93642 doxorubicin_02uM  3</span></span></code></pre></div>
<p>You can extract multiple chromosomes for multiple data:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="track-data-input.html#cb21-1" tabindex="-1"></a>hic_df <span class="ot">&lt;-</span> <span class="fu">prepareHic</span>(<span class="at">hic_path =</span> hic_data,</span>
<span id="cb21-2"><a href="track-data-input.html#cb21-2" tabindex="-1"></a>                     <span class="at">file_name =</span> <span class="fu">c</span>(<span class="st">&quot;doxorubicin_02uM&quot;</span>,<span class="st">&quot;doxorubicin_34uM&quot;</span>,</span>
<span id="cb21-3"><a href="track-data-input.html#cb21-3" tabindex="-1"></a>                                   <span class="st">&quot;ICRF193_5uM&quot;</span>),</span>
<span id="cb21-4"><a href="track-data-input.html#cb21-4" tabindex="-1"></a>                     <span class="at">chrom =</span> <span class="fu">c</span>(<span class="st">&quot;1&quot;</span>,<span class="st">&quot;2&quot;</span>,<span class="st">&quot;3&quot;</span>),<span class="at">assembly =</span> <span class="st">&quot;hg19&quot;</span>,</span>
<span id="cb21-5"><a href="track-data-input.html#cb21-5" tabindex="-1"></a>                     <span class="at">resolution =</span> <span class="dv">10000</span>)</span>
<span id="cb21-6"><a href="track-data-input.html#cb21-6" tabindex="-1"></a></span>
<span id="cb21-7"><a href="track-data-input.html#cb21-7" tabindex="-1"></a><span class="co"># check</span></span>
<span id="cb21-8"><a href="track-data-input.html#cb21-8" tabindex="-1"></a><span class="fu">head</span>(hic_df[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,])</span>
<span id="cb21-9"><a href="track-data-input.html#cb21-9" tabindex="-1"></a><span class="co">#   seqnames  start   end     score         fileName id</span></span>
<span id="cb21-10"><a href="track-data-input.html#cb21-10" tabindex="-1"></a><span class="co"># 1     chr1 710000  5000  72.90664 doxorubicin_02uM  1</span></span>
<span id="cb21-11"><a href="track-data-input.html#cb21-11" tabindex="-1"></a><span class="co"># 2     chr1 715000     0 323.53738 doxorubicin_02uM  2</span></span>
<span id="cb21-12"><a href="track-data-input.html#cb21-12" tabindex="-1"></a><span class="co"># 3     chr1 715000 10000 212.93642 doxorubicin_02uM  3</span></span></code></pre></div>
<p>Besides, you can define a region to extract:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="track-data-input.html#cb22-1" tabindex="-1"></a>hic_df <span class="ot">&lt;-</span> <span class="fu">prepareHic</span>(<span class="at">hic_path =</span> hic_data,</span>
<span id="cb22-2"><a href="track-data-input.html#cb22-2" tabindex="-1"></a>                     <span class="at">file_name =</span> <span class="fu">c</span>(<span class="st">&quot;doxorubicin_02uM&quot;</span>,<span class="st">&quot;doxorubicin_34uM&quot;</span>,</span>
<span id="cb22-3"><a href="track-data-input.html#cb22-3" tabindex="-1"></a>                                   <span class="st">&quot;ICRF193_5uM&quot;</span>),</span>
<span id="cb22-4"><a href="track-data-input.html#cb22-4" tabindex="-1"></a>                     <span class="at">chrom =</span> <span class="st">&quot;1&quot;</span>,<span class="at">assembly =</span> <span class="st">&quot;hg19&quot;</span>,</span>
<span id="cb22-5"><a href="track-data-input.html#cb22-5" tabindex="-1"></a>                     <span class="at">chromstart =</span> <span class="dv">20000000</span>, <span class="at">chromend =</span> <span class="dv">47500000</span>,</span>
<span id="cb22-6"><a href="track-data-input.html#cb22-6" tabindex="-1"></a>                     <span class="at">resolution =</span> <span class="dv">10000</span>)</span>
<span id="cb22-7"><a href="track-data-input.html#cb22-7" tabindex="-1"></a></span>
<span id="cb22-8"><a href="track-data-input.html#cb22-8" tabindex="-1"></a><span class="co"># check</span></span>
<span id="cb22-9"><a href="track-data-input.html#cb22-9" tabindex="-1"></a><span class="fu">head</span>(hic_df[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,])</span>
<span id="cb22-10"><a href="track-data-input.html#cb22-10" tabindex="-1"></a><span class="co">#   seqnames    start  end     score         fileName id</span></span>
<span id="cb22-11"><a href="track-data-input.html#cb22-11" tabindex="-1"></a><span class="co"># 1     chr1 19995000    0  96.48324 doxorubicin_02uM  1</span></span>
<span id="cb22-12"><a href="track-data-input.html#cb22-12" tabindex="-1"></a><span class="co"># 2     chr1 20000000 5000  38.00954 doxorubicin_02uM  2</span></span>
<span id="cb22-13"><a href="track-data-input.html#cb22-13" tabindex="-1"></a><span class="co"># 3     chr1 20005000    0 102.91180 doxorubicin_02uM  3</span></span></code></pre></div>
<hr />
<blockquote>
<p>Test data can be fetched on <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE222637"><strong>GSE222637</strong></a>.</p>
</blockquote>
<p>For <strong>.cool</strong> data format, <strong>prepareHic</strong> runs slowly and spend much memory:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="track-data-input.html#cb23-1" tabindex="-1"></a>hic_data <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;test-bw3/&quot;</span>,<span class="at">pattern =</span> <span class="st">&quot;.cool&quot;</span>,<span class="at">full.names =</span> T)</span>
<span id="cb23-2"><a href="track-data-input.html#cb23-2" tabindex="-1"></a>hic_data</span>
<span id="cb23-3"><a href="track-data-input.html#cb23-3" tabindex="-1"></a><span class="co"># [1] &quot;test-bw3/HiC_Adril-1_10kb.cool&quot; &quot;test-bw3/HiC_Ctrl-1_10kb.cool&quot;</span></span>
<span id="cb23-4"><a href="track-data-input.html#cb23-4" tabindex="-1"></a></span>
<span id="cb23-5"><a href="track-data-input.html#cb23-5" tabindex="-1"></a>hic_df <span class="ot">&lt;-</span> <span class="fu">prepareHic</span>(<span class="at">hic_path =</span> hic_data,</span>
<span id="cb23-6"><a href="track-data-input.html#cb23-6" tabindex="-1"></a>                     <span class="at">chrom =</span> <span class="st">&quot;chr1&quot;</span>,</span>
<span id="cb23-7"><a href="track-data-input.html#cb23-7" tabindex="-1"></a>                     <span class="at">resolution =</span> <span class="dv">10000</span>)</span>
<span id="cb23-8"><a href="track-data-input.html#cb23-8" tabindex="-1"></a></span>
<span id="cb23-9"><a href="track-data-input.html#cb23-9" tabindex="-1"></a><span class="co"># check</span></span>
<span id="cb23-10"><a href="track-data-input.html#cb23-10" tabindex="-1"></a><span class="fu">head</span>(hic_df,<span class="dv">3</span>)</span>
<span id="cb23-11"><a href="track-data-input.html#cb23-11" tabindex="-1"></a><span class="co">#   seqnames    start      end score         fileName id</span></span>
<span id="cb23-12"><a href="track-data-input.html#cb23-12" tabindex="-1"></a><span class="co"># 1     chr1    -5000        0     4 HiC_Adril-1_10kb  1</span></span>
<span id="cb23-13"><a href="track-data-input.html#cb23-13" tabindex="-1"></a><span class="co"># 2     chr1    80000    85000     1 HiC_Adril-1_10kb  2</span></span>
<span id="cb23-14"><a href="track-data-input.html#cb23-14" tabindex="-1"></a><span class="co"># 3     chr1 59660000 59665000     1 HiC_Adril-1_10kb  3</span></span></code></pre></div>
</div>
<div id="extract-junction-data" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> Extract junction data<a href="track-data-input.html#extract-junction-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>loadJunction</strong> can be used to load junctions data from your own <strong>bed</strong> format
data which records differential splice sites information from other tools identified
or from your <strong>bam</strong> file. The latter we use <strong>megadepth::bam_to_junctions</strong> to
extract all junctions data and return a data frame format. More details see
<a href="https://bioconductor.org/packages/release/bioc/vignettes/megadepth/inst/doc/megadepth.html#1_Basics"><strong>megadepth</strong></a>. Here we show examples:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="track-data-input.html#cb24-1" tabindex="-1"></a>bam_file <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">&quot;F:/junc-test/&quot;</span>,</span>
<span id="cb24-2"><a href="track-data-input.html#cb24-2" tabindex="-1"></a>                       <span class="at">pattern =</span> <span class="st">&quot;.bam$&quot;</span>,<span class="at">full.names =</span> T)</span>
<span id="cb24-3"><a href="track-data-input.html#cb24-3" tabindex="-1"></a>bam_file</span>
<span id="cb24-4"><a href="track-data-input.html#cb24-4" tabindex="-1"></a><span class="co"># [1] &quot;F:/junc-test/C1.sorted.bam&quot; &quot;F:/junc-test/WT.sorted.bam&quot;</span></span>
<span id="cb24-5"><a href="track-data-input.html#cb24-5" tabindex="-1"></a></span>
<span id="cb24-6"><a href="track-data-input.html#cb24-6" tabindex="-1"></a>junc_df <span class="ot">&lt;-</span> <span class="fu">loadJunction</span>(<span class="at">data_path =</span> bam_file,</span>
<span id="cb24-7"><a href="track-data-input.html#cb24-7" tabindex="-1"></a>                        <span class="at">file_name =</span> <span class="fu">c</span>(<span class="st">&quot;C1&quot;</span>,<span class="st">&quot;WT&quot;</span>))</span>
<span id="cb24-8"><a href="track-data-input.html#cb24-8" tabindex="-1"></a>junc_df <span class="ot">&lt;-</span> junc_df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(score <span class="sc">&gt;=</span> <span class="dv">5</span>)</span>
<span id="cb24-9"><a href="track-data-input.html#cb24-9" tabindex="-1"></a></span>
<span id="cb24-10"><a href="track-data-input.html#cb24-10" tabindex="-1"></a><span class="co"># check</span></span>
<span id="cb24-11"><a href="track-data-input.html#cb24-11" tabindex="-1"></a><span class="fu">head</span>(junc_df,<span class="dv">3</span>)</span>
<span id="cb24-12"><a href="track-data-input.html#cb24-12" tabindex="-1"></a><span class="co">#   seqnames   start     end score fileName</span></span>
<span id="cb24-13"><a href="track-data-input.html#cb24-13" tabindex="-1"></a><span class="co"># 1        1 3154117 3159706     1       C1</span></span>
<span id="cb24-14"><a href="track-data-input.html#cb24-14" tabindex="-1"></a><span class="co"># 2        1 3207318 3213608     1       C1</span></span>
<span id="cb24-15"><a href="track-data-input.html#cb24-15" tabindex="-1"></a><span class="co"># 3        1 4492669 4493099     8       C1</span></span></code></pre></div>
<p>The score stands for read count of junctions. A little time you will spend if you
extarct all junctions site from bam files. I recommend you featch the significant
junction sites information from other softwares for visualization.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="draw-chromosome-ideograms.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="basic-visualization.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-Track_data_input.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
